imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -255,6 +278,111 @@ def create_pset_counts(
     return counts
 
 
+def get_pointing_times(epoch: int) -> tuple[xr.DataArray, xr.DataArray]:
+    """
+    Get the pointing start and end times for a given epoch.
+
+    The pointing start time is the end of the repointing period
+    and the pointing end time is the start of the next repointing period.
+
+    Parameters
+    ----------
+    epoch : int
+        The first epoch time in the L1B Direct Events (TTJ2000NS).
+
+    Returns
+    -------
+    tuple[xarray.DataArray, xarray.DataArray]
+        The pointing start and end times in TTJ2000NS.
+        If no pointing times are found, both values will be None.
+    """
+    # Get the repoint table
+    repoint_df = get_repoint_data()
+
+    # Convert repoint and spin MET to TTJ2000NS to compare with the epoch
+    repoint_df["repoint_end_ttj2000ns"] = met_to_ttj2000ns(
+        repoint_df["repoint_end_met"]
+    )
+    repoint_df["repoint_start_ttj2000ns"] = met_to_ttj2000ns(
+        repoint_df["repoint_start_met"]
+    )
+
+    repoint_mask = (repoint_df["repoint_end_ttj2000ns"] <= epoch) & (
+        repoint_df["repoint_start_ttj2000ns"].shift(-1) >= epoch
+    )
+
+    # Get the repoint index that contains the epoch
+    repoint_idx = repoint_mask.idxmax() if repoint_mask.any() else None
+    # use the index to get the pointing start and end times
+    pointing_start = (
+        repoint_df["repoint_end_met"].iloc[repoint_idx]
+        if repoint_idx is not None
+        else None
+    )
+    pointing_end = (
+        repoint_df["repoint_start_met"].iloc[repoint_idx + 1]
+        if repoint_idx is not None
+        else None
+    )
+
+    return (
+        xr.DataArray(np.array([pointing_start]), dims=["epoch"]),
+        xr.DataArray(np.array([pointing_end]), dims=["epoch"]),
+    )
+
+
+def get_spin_numbers(
+    pointing_start: int, pointing_end: int
+) -> tuple[xr.DataArray, xr.DataArray]:
+    """
+    Get the start and end spin numbers from the L1B Direct Event dataset.
+
+    The spin numbers are used to identify the start and end of the
+    pointing periods in the dataset.
+
+    Parameters
+    ----------
+    pointing_start : int
+        The pointing start time in MET.
+    pointing_end : int
+        The pointing end time in MET.
+
+    Returns
+    -------
+    tuple[xarray.DataArray, xarray.DataArray]
+        The start and end spin numbers.
+    """
+    # Get the spin table
+    spin_df = get_spin_data()
+
+    # create a mask for the spin that contains the start of the pointing
+    mask_start = (spin_df["spin_start_met"] <= pointing_start) & (
+        spin_df["spin_start_met"].shift(-1) > pointing_start
+    )
+    # create a masks for the spin that contains the end of the pointing
+    mask_end = (spin_df["spin_start_met"] <= pointing_end) & (
+        spin_df["spin_start_met"].shift(-1) >= pointing_end
+    )
+    # get the index for the spin that contains the start and end of the pointing
+    spin_start_idx = mask_start.idxmax() if mask_start.any() else None
```

Comment:
Shouldn't this be the minimum value of the condition to get the start?
```suggestion
    spin_start_idx = mask_start.idxmin() if mask_start.any() else None
```